aiohue/v2/controllers/sensors.py

```diff
@@ -39,6 +42,65 @@ class ButtonController(BaseResourcesController[Type[Button]]):
 
     item_type = ResourceTypes.BUTTON
 
+    _workaround_tasks: Dict[str, asyncio.Task] = {}
+
+    async def _handle_event(self, type: EventType, item: Button | None) -> None:
+        """Handle incoming event for this resource from the EventStream."""
+        await super()._handle_event(type, item)
+
+        # Handle longpress workaround if needed
+        if (
+            not item
+            or type != EventType.RESOURCE_UPDATED
+            or not item.button
+            or item.button.last_event != ButtonEvent.INITIAL_PRESS
+        ):
+            return
+
+        device = self.get_device(item.id)
+        if device is None or device.product_data.model_id not in BTN_WORKAROUND_NEEDED:
+            return
+
+        if item.id in self._workaround_tasks:
+            # cancel existing task (if any)
+            # should not happen, but just in case
+            task = self._workaround_tasks.pop(item.id)
+            if not task.done():
+                task.cancel()
```

Comment:
The reason I used to await this task is that it is sending an event. So if the user has an automation listening to events from this remote, it's better that the last "sequence" finishes before the next event sequence comes in.

Reply:
I guess it doesn't matter because we are sleeping 1.5s in the workaround, so never end up firing an event earlier.

Reply:
that was my reasoning indeed